src/Symfony/Component/Process/Process.php

```diff
@@ -268,20 +268,16 @@ public function start(callable $callback = null, array $env = array())
             }
         }
 
-        if (null === $env) {
-            $env = $this->env;
-        } elseif ($this->env) {
+        if ($this->env) {
             $env += $this->env;
         }
 
         $envBackup = array();
-        if (null !== $env) {
-            foreach ($env as $k => $v) {
-                $envBackup[$k] = getenv($k);
-                putenv(false === $v || null === $v ? $k : "$k=$v");
-            }
-            $env = null;
+        foreach ($env as $k => $v) {
+            $envBackup[$k] = getenv($k);
+            putenv(false === $v || null === $v ? $k : "$k=$v");
         }
+        $env = null;
```

Comment:
Instead of overwriting `$env` here and using it in `proc_open` below, I suggest using `null` explicitly in the `proc_open` call. It would be more readable.

Reply:
Done.